lib/improver/blending/weighted_blend.py

```diff
@@ -106,12 +107,20 @@ def _rationalise_blend_time_coords(self, cubelist, cycletime=None):
             cycletime (str or None):
                 The cycletime in a YYYYMMDDTHHMMZ format e.g. 20171122T0100Z
 
+        Returns:
+            iris.cube.CubeList
+
         Raises:
             ValueError: if forecast_reference_time (to be unified) is a
                 dimension coordinate
         """
+        # create copies so as not to modify inputs
+        cubes = []
+        for cube in cubelist:
+            cubes.append(cube.copy())
```

Comment:
This isn't straightforward but my solution may not be the best one - let's discuss.

Reply:
OK so here are some pertinent things:

- `unify_forecast_reference_time` is also used in time-lagging, so it needs to be "safe" (ie return OR modify, preferably return as that's how it's being used there)
- It's possible we should make `unify_...` do the cycletime logic as it's identical in both places it's called
- `_rationalise_blend_time_coordinates` is doing a couple of different things.  If I move the cycletime logic / parsing into `unify_...`, I can then do things in a more sensible order from `MergeCubesForWeightedBlending.process` by breaking up / removing this function completely, and thus returning something with only its cycletime updated, for example, if there's only one cube in the list.  It also means there's only one `copy` going on implicitly in `unify_forecast_reference_time`, which we can rely on and note in a comment (ie get rid of the duplicated cube copying).  And I should add a unit test to cover that in case of future code updates.

So I think I should do the above and see what that does.